lib/auth/auth_with_roles.go

```diff
@@ -2198,9 +2201,11 @@ func (a *ServerWithRoles) GetDatabaseServers(ctx context.Context, namespace stri
 	}
 	// Filter out databases the caller doesn't have access to from each server.
 	var filtered []types.DatabaseServer
+	// MFA is not required to list the databases, but will be required to
+	// connect to them.
+	mfaVerified := true
```

Comment:
I'm not sure how you plan to propagate this flag but I was wondering whether the checker value could be 
constructed with this attribute built-in instead of threading it in each case?
I understand this could challenging in case `RoleSet` as it does not carry additional context currently.

Reply:
This flag is already propagated in all the necessary places in this PR.
Embedding it in a `RoleSet` is doable but this would change what that type represents. An alternative is to wrap `services.AccessChecker` but the methods exposed there wouldn't work for the MFA use case.

I prefer to keep the existing type semantics instead of avoiding some flag plumbing.
WDYT?

Reply:
I agree that it's counter-productive. Yet the reason I brought it up was that `mfaVerified` attribute applies to the whole access checker (e.g. roleset) regardless of its state and it's my understanding that parameters are used for volatile and dynamic input (which mfaVerified isn't) - hence my question.